ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Background.hs

```diff
@@ -82,10 +82,15 @@ launchBgTasks cdb@CDB{..} = do
       copyToImmDBRunner cdb gcSchedule
     lgrSnapshotThread <- launch $
       updateLedgerSnapshotsRunner cdb
+    -- TODO: Do we still need this cdbBgThreads, if we have the registry?
```

Comment:
Hmm, I think the only reason we have `cdbBgThreads` is to be able to shut down the background threads before we close the ImmutableDB and the VolatileDB, so that no background thread tries to access them after their closure, which would result in a `ClosedDBError`.

Since the ImmutableDB and the VolatileDB are not managed by the `ResourceRegistry` (maybe they should be?!), these actions do not take place in the right order, I believe.

Reply:
I'll leave it for now. It kinda feels to me that this `MVar` is a registry in disguise, and so maybe we should just use one, but it's not essential to this PR.